internal/policies/proxy/proxy_test.go

```diff
@@ -0,0 +1,136 @@
+package proxy_test
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"os"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/adsys/internal/policies/entry"
+	"github.com/ubuntu/adsys/internal/policies/proxy"
+	"github.com/ubuntu/adsys/internal/testutils"
+)
+
+func TestApplyPolicy(t *testing.T) {
+	t.Cleanup(testutils.StartLocalSystemBus())
+	t.Parallel()
+
+	bus := testutils.NewDbusConn(t)
+	tests := map[string]struct {
+		entries []entry.Entry
+
+		isUser        bool
+		dbusCallError bool
+
+		wantErr       bool
+		wantApplyArgs []string
+	}{
+		// Computer cases
+		"Computer, no entries":                   {},
+		"Computer, no entries, D-Bus call error": {dbusCallError: true},
+		"Computer, single enabled entry": {
+			entries:       []entry.Entry{{Key: "proxy/auto", Value: "http://example.com:8080/proxy.pac"}},
+			wantApplyArgs: []string{"", "", "", "", "", "http://example.com:8080/proxy.pac"},
+		},
+		"Computer, single disabled entry": {
+			entries:       []entry.Entry{{Key: "proxy/http", Value: "", Disabled: true}},
+			wantApplyArgs: []string{"", "", "", "", "", ""},
+		},
+		"Computer, all entries set": {
+			entries: []entry.Entry{
+				{Key: "proxy/http", Value: "http://example.com:8080"},
+				{Key: "proxy/https", Value: "https://example.com:8080"},
+				{Key: "proxy/ftp", Value: "ftp://example.com:8080"},
+				{Key: "proxy/socks", Value: "socks://example.com:8080"},
+				{Key: "proxy/no-proxy", Value: "localhost,127.0.0.1"},
+				{Key: "proxy/auto", Value: "http://example.com:8080/proxy.pac"},
+			},
+			wantApplyArgs: []string{
+				"http://example.com:8080",
+				"https://example.com:8080",
+				"ftp://example.com:8080",
+				"socks://example.com:8080",
+				"localhost,127.0.0.1",
+				"http://example.com:8080/proxy.pac",
+			},
+		},
+
+		// User cases
+		"User, no entries":        {isUser: true},
+		"User, non-empty entries": {isUser: true, entries: []entry.Entry{{Key: "not-applied", Value: "not-applied"}}},
+
```

Comment:
I don’t see a test case with no error (we won’t test the warning), when getting errDBusServiceUnknownName. Is that line covered? (I only see the one in the integration tests)

Reply:
you’re right, force pushing makes reviewing the diff of diff harder :) I was expecting it to be in that table testing. Fine by me then, let’s approve. Thanks!